FinalEngine.Editor.Common/Services/Application/ApplicationContext.cs

```diff
@@ -5,14 +5,63 @@
 namespace FinalEngine.Editor.Common.Services.Application;
 
 using System;
+using System.IO.Abstractions;
 using System.Reflection;
+using FinalEngine.Editor.Common.Services.Environment;
 
 /// <summary>
 /// Provides a standard implementation of an <see cref="IApplicationContext"/>.
 /// </summary>
 /// <seealso cref="IApplicationContext" />
 public sealed class ApplicationContext : IApplicationContext
 {
+    /// <summary>
+    /// The environment service, used when locating the applications local data directory.
+    /// </summary>
+    private readonly IEnvironmentContext environment;
+
+    /// <summary>
+    /// The file system service, used to potentially create the applications local data directory.
+    /// </summary>
+    private readonly IFileSystem fileSystem;
```

Comment:
I didn't know that the constructor would take care of the documentation. I do feel like it can hold merit for developers though, particularly when fields are used as backing fields for properties.